src/python/pants/backend/adhoc/target_types.py

```diff
@@ -345,6 +345,18 @@ class SystemBinaryFingerprintDependenciesField(AdhocToolRunnableDependenciesFiel
         """
     )
 
+class SystemBinaryExtraEnvVarsField(StringSequenceField):
+    alias: ClassVar[str] = "extra_env_vars"
+    help = help_text(
+        """
+        Additional environment variables to provide to the system binary during fingerprinting.
+        This has no effect on the execution of the binary in the scope of an `adhoc_tool` or
```

Comment:
Hmm, I think these env vars should always be set, not just for fingerprinting (although the ad-hoc tool values should override them if specified). In your case it would mean you could set DISPLAY in just one place.

Reply:
Thanks for the feedback. Need clarification.

1. Let's assume an `adhoc_tool` has multiple `runnable_dependencies`. One of them sets `extra_env_vars` in the `system_binary`. That means the `adhoc_tool` invocation would set the vars for the whole invocation, means for the `runnable` and all other binaries. Is this a safe assumption?
2. If yes, I would combine *all* `extra_env_vars` of *all* `runnable_dependencies`. This may also conflict, e.g. one sets an env var to take it over from pants env, the other binary defines it explicitly.  How to deal with this?
3. If, however, the `adhoc_tool` defines `extra_env_vars` I would only consider this field.

I think it comes down to what we think the user wants.

Reply:
I thought about that again and I think you're right.
If a variable is needed for fingerprinting, it most certainly is also needed for running the tool. So I will merge all env var requests of binaries.

Reply:
Yeah, I think merge them in order of specificity, so the more specific targets overrides the less specific one.